sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/DcosVersion.java

```diff
@@ -84,4 +120,22 @@ private static String getVersionElement(String version, int index) {
         }
         return elements[index];
     }
+
+    private static DcosVariant parseVariant(JSONObject jsonObject) {
+        DcosVariant variant = DcosVariant.UNKNOWN;
+        if (jsonObject.has(VARIANT_KEY)) {
+            String variantStr = jsonObject.getString(VARIANT_KEY);
+
+            if (variantStr.equals(DcosVariant.OPEN.toString())) {
```

Comment:
I preferred this over `DcosVariant.valueOf(parsedString.toUpperCase(Locale.ENGLISH))`. Let me know if you have a better suggestion.

Reply:
I do not.

But, I would put this method inside the DcosVariant enum.